store/postgres/replay_repository.go

```diff
@@ -19,57 +22,112 @@ type Replay struct {
 	JobID uuid.UUID `gorm:"not null"`
 	Job   Job       `gorm:"foreignKey:JobID"`
 
-	StartDate time.Time `gorm:"not null"`
-	EndDate   time.Time `gorm:"not null"`
-	Status    string    `gorm:"not null"`
-	Message   datatypes.JSON
+	StartDate     time.Time `gorm:"not null"`
+	EndDate       time.Time `gorm:"not null"`
+	Status        string    `gorm:"not null"`
+	Message       datatypes.JSON
+	ExecutionTree datatypes.JSON
 
 	CreatedAt time.Time `gorm:"not null" json:"created_at"`
 	UpdatedAt time.Time `gorm:"not null" json:"updated_at"`
 }
 
+type ExecutionTree struct {
+	Data       interface{}
+	Dependents []*ExecutionTree
+	Runs       []time.Time
+}
+
+func fromTreeNode(executionTree *tree.TreeNode) *ExecutionTree {
+	var dependents []*ExecutionTree
+	for _, dependent := range executionTree.Dependents {
+		dependents = append(dependents, fromTreeNode(dependent))
+	}
+
+	var runs []time.Time
+	for _, run := range executionTree.Runs.Values() {
+		runs = append(runs, run.(time.Time))
+	}
+
+	return &ExecutionTree{
+		Data:       executionTree.Data.(models.JobSpec),
+		Dependents: dependents,
+		Runs:       runs,
+	}
+}
+
 func (p Replay) FromSpec(spec *models.ReplaySpec) (Replay, error) {
-	jsonBytes, err := json.Marshal(spec.Message)
+	message, err := json.Marshal(spec.Message)
 	if err != nil {
 		return Replay{}, nil
 	}
+
+	var executionTree []byte
+	if spec.ExecutionTree != nil {
+		executionTree, err = json.Marshal(fromTreeNode(spec.ExecutionTree))
+		if err != nil {
+			return Replay{}, err
+		}
+	}
+
 	return Replay{
-		ID:        spec.ID,
-		JobID:     spec.Job.ID,
-		StartDate: spec.StartDate.UTC(),
-		EndDate:   spec.EndDate.UTC(),
-		Status:    spec.Status,
-		Message:   jsonBytes,
+		ID:            spec.ID,
+		JobID:         spec.Job.ID,
+		StartDate:     spec.StartDate.UTC(),
+		EndDate:       spec.EndDate.UTC(),
+		Status:        spec.Status,
+		Message:       message,
+		ExecutionTree: executionTree,
 	}, nil
 }
 
+func toTreeNode(executionTree *ExecutionTree) *tree.TreeNode {
+	var jobSpec models.JobSpec
+	mapstructure.Decode(executionTree.Data, &jobSpec)
```

Comment:
why are we ignoring error returned by this `Decode`?

Reply:
refactored to not use mapstructure